app/jobs/notifications/remove_all_by_action_job.rb

```diff
@@ -0,0 +1,9 @@
+module Notifications
+  class RemoveAllByActionJob < ApplicationJob
+    queue_as :remove_all_by_action_notifications
+
+    def perform(notifiable_id, notifiable_type, action, service = Notifications::RemoveAllByAction)
```

Comment:
Since this takes an array of IDs the name should probably be `notifiable_ids`?

You can make sure it's an array by using `Array.wrap(notifiable_ids)`, in case someone passes a single ID without it being wrapped in an array

Reply:
we should also probably guard what type of `notifiable_type` can be passed, so that the job can't just load "whatever" from the DB

Something like:

`return unless %w[Article Comment].include(notifiable_type)` 

I guess?

Reply:
TIL `Array.wrap`! Great method.

Reply:
Hmm, @rhymes @lightalloy thoughts on where these guard statements should be? Should they be in the job, the class, or the model level `notification.rb`? Might be overthinking it though.

Reply:
That's a fair question. Since this is a condition that will be checked every single time, I think it makes sense to keep it in the job to reduce duplicate codes.